src/main/java/org/broadinstitute/hellbender/utils/fermi/FermiLiteAssembler.java

```diff
@@ -0,0 +1,127 @@
+package org.broadinstitute.hellbender.utils.fermi;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.NativeUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class FermiLiteAssembler implements Assembler {
+    private static volatile boolean nativeLibLoaded = false;
+
+    public FermiLiteAssembler() {
+        loadNativeLibrary();
+    }
+
+    public Assembly createAssembly( final Collection<GATKRead> reads ) {
+        final ByteBuffer assemblyData = createAssemblyData(makeReadData(reads));
+        if ( assemblyData == null ) throw new IllegalStateException("Unable to create assembly.");
+        try {
+            return interpretAssemblyData(assemblyData);
+        } finally {
+            destroyAssemblyData(assemblyData);
+        }
+    }
+
+    private static void loadNativeLibrary() {
+        if ( !nativeLibLoaded ) {
+            synchronized (FermiLiteAssembler.class) {
+                if ( !nativeLibLoaded ) {
+                    final String libNameOverride = System.getProperty("LIBFML_PATH");
+                    final String libName;
+                    if ( libNameOverride != null ) libName = libNameOverride;
+                    else if ( NativeUtils.runningOnPPCArchitecture() ) libName = null;
+                    else if (NativeUtils.runningOnMac()) libName = "/libfml.Darwin.dylib";
+                    else if (NativeUtils.runningOnLinux()) libName = "/libfml.Linux.so";
+                    else libName = null;
+                    if ( libName == null ) {
+                        throw new UserException.HardwareFeatureException("We have a JNI binding for fermi-lite only for Linux and Mac.");
+                    }
+                    if ( !NativeUtils.loadLibraryFromClasspath(libName) ) {
+                        throw new UserException.HardwareFeatureException("Misconfiguration: Unable to load fermi-lite native library "+libName);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
I checked with the BwaMem binding PR (`BwaMemIndex.java` line 338), there is a write at the end of the synchronized call.